Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoASA.yaml

```diff
@@ -0,0 +1,576 @@
+                | where AliasDeviceEventClassID == "106023" and not(Message has "protocol 41")
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * ' by access-group "' NetworkRuleName '" '  *
+                | parse Message with * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")" *;
+      let all_106023_unparsed = unparsedData
+                | where AliasDeviceEventClassID == "106023" and not(Message has "protocol 41")
+                | parse Message with * ":" DeviceAction " " Protocol " src " SrcInterfaceName ":" SrcIpAddrAndPort "(" SrcUsername ") dst " DstInterfaceName ":" DstIpAddrAndPort " " NetworkIcmpInfo 'by access-group "' NetworkRuleName '" [' * "]"
+                | parse NetworkIcmpInfo with "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ") "
+                | extend SrcIpAddrAndPort = split(SrcIpAddrAndPort,"/"), DstIpAddrAndPort = split(DstIpAddrAndPort,"/")
+                | extend SrcIpAddr = tostring(SrcIpAddrAndPort[0]),
+                        SrcPortNumber = toint(SrcIpAddrAndPort[1]),
+                        DstIpAddr = tostring(DstIpAddrAndPort[0]),
+                        DstPortNumber = toint(DstIpAddrAndPort[1]),
+                        DeviceAction = trim_start(@"\s+",DeviceAction)
+                | extend 
+                        SimplifiedDeviceAction = iff(isempty(SimplifiedDeviceAction), 
+                                                        DeviceAction, 
+                                                        SimplifiedDeviceAction)
+                | where (isnull(dstportnumber) or (DstPortNumber == dstportnumber))
+                | extend temp_isSrcMatch=has_any_ipv4_prefix(SrcIpAddr,src_or_any), 
+                                temp_isDstMatch=has_any_ipv4_prefix(DstIpAddr,dst_or_any)
+                | extend ASimMatchingIpAddr = case(
+                                                array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-", // match not requested
+                                                (temp_isSrcMatch and temp_isDstMatch), "Both", // has to be checked before the individual 
+                                                temp_isSrcMatch, "SrcIpAddr",
+                                                temp_isDstMatch, "DstIpAddr",
+                                                "No match"
+                                            )
+                | where ASimMatchingIpAddr != "No match"
+                | where ((array_length(dvcaction) == 0) or SimplifiedDeviceAction has_any (dvcaction))
+                | where ((eventresult == "*") or 
+                        (eventresult == "Success" and SimplifiedDeviceAction == "Allow") or 
+                        (eventresult == "Failure" and SimplifiedDeviceAction != "Allow"))
+                | project-away SrcIpAddrAndPort, DstIpAddrAndPort, NetworkIcmpInfo;
+      let all_106023_41 = union parsedData, unparsedData
```

Comment:
Just noticed the parsed/unparsed union: you might have got it right in the ASIM parser - though it certainly duplicates work - but it is still very hard to maintain this "reverse-fork" in the parser. Given the challenges of the vim parsers, this would just not work.

Reply:
Only unparsed is required. Updated.